### PR TITLE
Feature/23 email validation

### PR DESCRIPTION
**Pull request by fasbrothers**

### CODE REVIEW

**Review by mirsadikov — APPROVED** [review]

File: src/api/mutations/auth/useVerifyEmailMutation.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const navigate = useNavigate();
+
+  return useMutation({
+    mutationKey: ['verify-email'],
+    mutationFn: async () => {
+      await httpClient.post<string>(VERIFY_EMAIL_ROUTE, { token });
+    },
+    onSuccess: () => {
+      toast.success('Email verified');
+      navigate('/auth/signin');
```
Comment (otabek) [inline src/api/mutations/auth/useVerifyEmailMutation.ts]: can you also add this route to src/api/routes.ts

File: src/api/routes.ts [hunk 2]
```diff
@@ -1 +1,2 @@
 export const AMENITIES_ROUTE = '/amenities';
+export const VERIFY_EMAIL_ROUTE = '/auth/verify';
```
Comment (otabek) [inline src/api/routes.ts]: better to separate if more than one route 

```
export const AUTH_ROUTE = '/auth';
export const VERIFY_EMAIL_ROUTE = `${AUTH_ROUTE}/verify`;
```

File: src/pages/auth/VerifyEmail.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { Box, Typography } from '@mui/material';
+import { useSearchParams } from 'react-router-dom';
+import LoadingPrimary from '../../components/LoadingPrimary';
+import useVerifyEmailMutation from '../../api/mutations/auth/useVerifyEmailMutation';
+
+function VerifyEmail() {
+  const [searchParams] = useSearchParams();
+  const token = searchParams.get('token') || '';
+
+  const { isPending, mutate } = useVerifyEmailMutation(token);
```
Comment (otabek) [inline src/pages/auth/VerifyEmail.tsx:11]: what happens if it fails to verify the email?

**Review by otabek — CHANGES_REQUESTED** [review]

File: src/components/mainNavigationStyles.ts
Comment (otabek) [inline src/components/mainNavigationStyles.ts:1]: stick to one file naming convention and create folder for each component 
```
components/navigation/MainNavigation.tsx
components/navigation/MainNavigation.styles.ts
```

**Review by otabek — CHANGES_REQUESTED** [review]

**Review by MdRasulov — APPROVED** [review]

**Review by otabek — APPROVED** [review]

**Pull request merged**